### PR TITLE
fix: remove minimum size requirements

### PR DESCRIPTION
**Pull request by i-chvets**
https://github.com/canonical/minio-operator/issues/78

Summary of changes:
- Removed minimum size requirements from metadata.yaml

NOTE: Extra changes (in addition to one in `metadata.yaml`) are required to have update libraries and test operator framework.

### CODE REVIEW

**Comment by ca-scribner** [comment]
What is the behaviour when there is no minimum size set?  Is it that there is no storage created, or that it just uses some juju default?  

I wonder if the best fix to #78 would be to remove the storage entirely.  Is that what happens here?

File: tox.ini [hunk 1]
```diff
@@ -50,10 +50,10 @@ description = Apply coding style standards to code
 commands =
     # uncomment the following line if this charm owns a lib
     # codespell {[vars]lib_path}
-    codespell {toxinidir}/. --skip {toxinidir}/.git --skip {toxinidir}/.tox \
-      --skip {toxinidir}/build --skip {toxinidir}/lib --skip {toxinidir}/venv \
-      --skip {toxinidir}/.mypy_cache \
-      --skip {toxinidir}/icon.svg --skip *.json.tmpl
+    codespell {toxinidir}/. --skip {toxinidir}/./.git --skip {toxinidir}/./.tox \
```
Comment (DnPlas) [inline tox.ini:53]: Can you explain what issue you are having without this change?

Comment (i-chvets) [reply tox.ini:53]: Those directories are not ignored by `codespell` if `/./` is not there. This is pretty much how it is done in other repos as well.

**Comment by i-chvets** [comment]
> 



> 

Storage is optional. No default minimum size is defined, as far as I can see. Charm is using `/data` directory as specified by storage, so I don't think it is safe to completely remove.

File: tox.ini (hunk 1 above)
Comment (ca-scribner) [reply tox.ini:53]: I suggest splitting this and the library bump up into a separate PR so it is clear what fixes #78 and what is maintenance

**Comment by ca-scribner** [comment]
If we cannot remove the storage entirely, what happens when we just omit the minimum size?  What size volume is being created?  I'm partly worried that, by not setting 10GB as the minimum, we're actually getting a larger volume

**Comment by i-chvets** [comment]
Looks like without minimum PVC is 1G:
`minio-data-f8ff2d4c-minio-0   Bound    pvc-955d1a2b-7702-45d8-a5c5-3549adee5702   1Gi        RWO            microk8s-hostpath`

File: tox.ini (hunk 1 above)
Comment (DnPlas) [reply tox.ini:53]: +1 on the above suggestion, please split this PR with those maintenance/CI fixes.

Comment (i-chvets) [reply tox.ini:53]: Got it. Will do.

**Pull request closed without merging**